api/routers/export.py

```diff
@@ -0,0 +1,169 @@
+import os
+import re
+import tempfile
+import logging
+
+from datetime import date, datetime
+from fastapi import APIRouter, HTTPException, BackgroundTasks
+from fastapi.responses import FileResponse
+from starlette import status
+from typing import List, Tuple
+from zipfile import ZipFile, ZIP_DEFLATED
+
+from api.models.export import ExportDTO, ExportDataType
+from api.utils import extract_config, clean_up_file
+from libs.metrics import FaceMaskUsageMetric, OccupancyMetric, SocialDistancingMetric
+
+logger = logging.getLogger(__name__)
+
+export_router = APIRouter()
+
+# Define the exports data types as constants
+ALL_DATA = ExportDataType.all_data
+RAW_DATA = ExportDataType.raw_data
+SOCIAL_DISTANCING = ExportDataType.social_distancing
+FACEMASK_USAGE = ExportDataType.facemask_usage
+OCCUPANCY = ExportDataType.occupancy
+
+
+def export_folder_into_zip(source_path, destination_path, zip_file, from_date, to_date):
+    if not os.path.exists(source_path):
+        return None
+    for filename in os.listdir(source_path):
+        if filename.endswith(".csv"):
+            if from_date and to_date:
+                # Date range was specified, only include files in that range
+                date_matches = re.findall(r"([0-9]{4}\-[0-9]{2}\-[0-9]{2})", filename)
+                if not date_matches:
+                    continue
+                file_date = datetime.strptime(date_matches[0], '%Y-%m-%d').date()
+                if not from_date <= file_date <= to_date:
+                    continue
+            zip_file.write(os.path.join(source_path, filename), arcname=os.path.join(destination_path, filename))
+
+
+def get_areas(export_info: ExportDTO) -> List[Tuple[str, str]]:
```

Comment:
:thinking: these methods don't retrieve the area's (or camera) data. They return a list with the (id, name) of the requested areas and cameras.

Reply:
These functions are only used in this file, right? In that case I on't think it is toobad to repeat function names. I think when a function is internal to a file we could prefix them with _

Reply:
Yeah my bad, I think after that many code reviews yesterday I got mixed up :sweat_smile:
Either an underscore or a name as `get_areas_from_export_info` would be good for me

Reply:
I renamed the functions as `get_areas_to_export` and  `get_cameras_to_export`.
I agree with the usage of the `_`. However, it's not used in the rest of the repository and I think that will be strange use it in a single file.

The name `get_areas_from_export_info` doen't work for the cameras because it also returns the cameras included in the selected areas.